src/cockpit/389-console/src/lib/ldap_editor/wizards/operations/addUser.jsx

```diff
@@ -39,26 +32,82 @@ class AddUser extends React.Component {
-            'preferredDeliveryMethod', 'displayName', 'employeeNumber', 'preferredLanguage'
-        ];
+        // Objectclasses
+        this.userObjectclasses = {
+            'Basic Account': [
+                'objectclass: top',
+                'objectClass: nsPerson',
+                'objectClass: nsAccount',
+                'objectClass: nsOrgPerson',
+            ],
+            'Posix Account': [
+                'objectclass: top',
+                'objectClass: nsPerson',
+                'objectClass: nsAccount',
+                'objectClass: nsOrgPerson',
+                'objectClass: posixAccount',
+            ],
+            'Service Account': [
+                'objectclass: top',
+                'objectClass: applicationProcess',
+                'objectClass: nsAccount'
+            ],
+        };
 
-        this.requiredAttributes = ['cn', 'sn'];
+        this.allowedAttrs = {
+            'Basic Account': [
+                'businessCategory', 'carLicense', 'departmentNumber',
+                'description', 'employeeNumber', 'employeeType', 'homePhone',
+                'homePostalAddress', 'initials', 'jpegPhoto', 'labeledURI',
+                'legalName', 'mail', 'manager', 'mobile', 'nsRoleDN', 'o',
+                'pager', 'photo', 'preferredLanguage', 'nsCertSubjectDN',
+                'nsSshPublicKey', 'roomNumber', 'seeAlso', 'telephoneNumber',
+                'uid', 'userCertificate', 'userPassword', 'userSMIMECertificate',
+                'userPKCS12', 'x500UniqueIdentifier'
+            ],
+            'Posix Account': [
+                'businessCategory', 'carLicense', 'departmentNumber',
+                'description', 'employeeNumber', 'employeeType', 'homePhone',
+                'homePostalAddress', 'initials', 'jpegPhoto', 'labeledURI',
+                'legalName', 'mail', 'manager', 'mobile', 'nsRoleDN', 'o',
+                'pager', 'photo', 'preferredLanguage', 'nsCertSubjectDN',
+                'nsSshPublicKey', 'roomNumber', 'seeAlso', 'telephoneNumber',
+                'userCertificate', 'userPassword', 'userSMIMECertificate',
+                'userPKCS12', 'x500UniqueIdentifier', 'loginShell', 'gecos'
+            ],
+            'Service Account': [
+                'seeAlso', 'ou', 'l', 'description', 'userCertificate',
+                'nsCertSubjectDN', 'nsRoleDN', 'nsSshPublicKey', 'userPassword'
+            ],
+        };
+
+        this.requiredAttrs = {
+            'Basic Account': [
+                'cn', 'displayName',
+            ],
+            'Posix Account': [
+                'cn', 'uid', 'uidNumber', 'gidNumber', 'homeDirectory',
+                'displayName'
+            ],
+            'Service Account': [
+                'cn'
+            ]
+        };
+
+        this.singleValuedAttrs = {
+            'Basic Account': [
+                'preferredDeliveryMethod', 'displayName', 'employeeNumber',
+                'preferredLanguage', 'userPassword',
```

Comment:
'userPassword' is not a single-valued attribute according to the default 00core.ldif schema file.

Reply:
Thanks for catching all those typo and copy-and paste errors!!!  As for userPassword, the server will only process the first userpassword it finds.  So essentially it is single valued, and I don't want people misusing the UI and thinking that can correctly add multiple passwords per entry

Reply:
Fixed @tmihinto